src/common/meta/src/rpc/router.rs

```diff
@@ -98,6 +102,48 @@ pub struct TableRoute {
     region_leaders: HashMap<RegionNumber, Option<Peer>>,
 }
 
+pub fn find_leaders(region_routes: &[RegionRoute]) -> HashSet<Peer> {
+    region_routes
+        .iter()
+        .flat_map(|x| &x.leader_peer)
+        .cloned()
+        .collect()
+}
+
+pub fn find_leader_regions(region_routes: &[RegionRoute], datanode: &Peer) -> Vec<RegionNumber> {
+    region_routes
+        .iter()
+        .filter_map(|x| {
+            if let Some(peer) = &x.leader_peer {
+                if peer == datanode {
+                    return Some(x.region.id.region_number());
+                }
+            }
+            None
+        })
+        .collect()
+}
+
+pub fn all_peers(region_routes: &[RegionRoute]) -> Vec<Peer> {
+    let mut peers = HashSet::new();
+    region_routes
+        .iter()
+        .filter_map(|x| x.leader_peer.as_ref())
+        .for_each(|p| {
+            let _ = peers.insert(p.clone());
+        });
+    region_routes
+        .iter()
+        .flat_map(|x| x.follower_peers.iter())
+        .for_each(|p| {
+            let _ = peers.insert(p.clone());
+        });
+    let mut peers = peers.into_iter().collect::<Vec<_>>();
+    peers.sort_by_key(|x| x.id);
```

Comment:
Why do we need sorting?

Reply:
Reduces unnecessary updates.  e.g., `[3, 1] != [1, 3]` 

We update the `DatanodeTableKey` like the following: 
```rust 
        for (datanode, regions) in new_region_distribution.into_iter() {
            if let Some(current_region) = current_region_distribution.get(&datanode) {
                // Updates if need.
                if *current_region != regions {
                    let key = DatanodeTableKey::new(datanode, table_id);
                    let raw_key = key.as_raw_key();
                    let val = DatanodeTableValue::new(table_id, regions).try_as_raw_value()?;
                    opts.push(TxnOp::Put(raw_key, val));
                }
            } else {
                // New datanodes
                let key = DatanodeTableKey::new(datanode, table_id);
                let raw_key = key.as_raw_key();
                let val = DatanodeTableValue::new(table_id, regions).try_as_raw_value()?;
                opts.push(TxnOp::Put(raw_key, val));
            }
        }
```